README.rst

```diff
@@ -0,0 +1,325 @@
+the previous MapReduce program (in our case ``PurchaseEventParser``) can be retrieved in this method using
+``workflowContext`` object. We query for the number of input records to the mapper and number of records
+emitted by the mapper. If at least 80% of the records were processed by the mapper, the method returns true,
+in which case the `fork` in the Workflow will get executed. If the method returns false, ``otherwise`` section in
+the ``condition`` gets executed, which contains ``EmailNotifier`` custom action.
+
+
+Build and Run Application
+=========================
+
+The ``PurchaseWorkflowApp`` can be built and packaged using the Apache Maven command::
+
+  $ mvn clean package
+
+Note that the remaining commands assume that the ``cdap-cli.sh`` script is
+available on your PATH. If this is not the case, please add it::
+
+  $ export PATH=$PATH:<CDAP home>/bin
+
+If you haven't already started a standalone CDAP installation, start it with the command::
+
+  $ cdap.sh start
+
+We can then deploy the application to the standalone CDAP installation::
+
+  $ cdap-cli.sh deploy app target/cdap-workflow-guide-<version>.jar
+
+Next, we will send some sample purchase events into the stream
+for processing::
+
+  $ cdap-cli.sh send stream purchaseEvents \'bob bought 3 apples for \$30\'
+  $ cdap-cli.sh send stream purchaseEvents \'joe bought 1 apple for \$100\'
+  $ cdap-cli.sh send stream purchaseEvents \'joe bought 10 pineapples for \$20\'
+  $ cdap-cli.sh send stream purchaseEvents \'cat bought 3 bottles for \$12\'
+  $ cdap-cli.sh send stream purchaseEvents \'cat bought 2 pops for \$14\'
+
+We can now start the Workflow program to process the events that were
+ingested::
+
+  $ cdap-cli.sh start workflow PurchaseWorkflowApp.PurchaseWorkflow
+
+The Workflow program will take a couple of minutes to execute.
+
+We can then start the ``PurchaseResultService`` and then query the processed
+results::
+
+  $ cdap-cli.sh start service PurchaseWorkflowApp.PurchaseResultService
+
+- Get purchase records for user ``joe``::
+
+  $ curl http://localhost:10000/v3/namespaces/default/apps/PurchaseWorkflowApp/services/PurchaseResultService/methods/purchaserecords/joe
+
+Example output::
+
+  [{"customer":"joe","product":"pineapple","quantity":10,"price":20,"purchaseTime":1430962917227},{"customer":"joe","product":"apple","quantity":1,"price":100,"purchaseTime":1430962917227}]
+
+- Get the total purchases made by user ``joe``::
+
+  $ curl http://localhost:10000/v3/namespaces/default/apps/PurchaseWorkflowApp/services/PurchaseResultService/methods/purchases/users/joe
+
+Example output::
+
+  120
+
+- Get the total purchases made for product ``apple``::
+
+  $ curl http://localhost:10000/v3/namespaces/default/apps/PurchaseWorkflowApp/services/PurchaseResultService/methods/purchases/products/apple
+
+Example output::
+
+  160
+
+You have now learned how to write a Workflow program to connect different MapReduce programs and run them in parallel
```

Comment:
You have now seen how to write a Workflow program to connect different MapReduce programs and run them in parallel based on a condition.